### PR TITLE
Replace gendered pronouns with gender neutral ones.

### PR DESCRIPTION
**Pull request by ghost**
https://github.com/systemd/systemd/issues/48

### CODE REVIEW

**Comment by NatsumiHB** [comment]
I'm in favor of this, inclusive speech is important nowadays

File: src/core/execute.c [hunk 1]
```diff
@@ -412,7 +412,7 @@ static int acquire_path(const char *path, int flags, mode_t mode) {
 
         if (connect(fd, &sa.sa, sa_len) < 0)
                 return errno == EINVAL ? -ENXIO : -errno; /* Propagate initial error if we get EINVAL, i.e. we have
-                                                           * indication that his wasn't an AF_UNIX socket after all */
+                                                           * indication that their wasn't an AF_UNIX socket after all */
```
Comment (michaelolbrich) [inline src/core/execute.c]: I think the 'his' was a typo here. This should be 'this', right?

File: src/udev/udev-builtin-net_id.c [hunk 2]
```diff
@@ -186,7 +186,7 @@ static int dev_pci_onboard(sd_device *dev, struct netnames *names) {
         /* Some BIOSes report rubbish indexes that are excessively high (2^24-1 is an index VMware likes to
          * report for example). Let's define a cut-off where we don't consider the index reliable anymore. We
          * pick some arbitrary cut-off, which is somewhere beyond the realistic number of physical network
-         * interface a system might have. Ideally the kernel would already filter his crap for us, but it
+         * interface a system might have. Ideally the kernel would already filter its crap for us, but it
```
Comment (michaelolbrich) [inline src/udev/udev-builtin-net_id.c]: I think it was supposed to be 'this' as well here.

**Comment by PhoenixDiscord** [comment]
I'll correct these in a second.

**Comment by PhoenixDiscord** [comment]
Done.

File: docs/JOURNAL_FILE_FORMAT.md [hunk 3]
```diff
@@ -196,7 +196,7 @@ The currently used part of the file is the **header_size** plus the
 **arena_size** field of the header. If a writer needs to write to a file where
 the actual file size on disk is smaller than the reported value it shall
 immediately rotate the file and start a new one. If a writer is asked to write
-to a file with a header that is shorter than his own definition of the struct
+to a file with a header that is shorter than their own definition of the struct
 Header, he shall immediately rotate the file and start a new one.
```
Comment (keszybz) [inline docs/JOURNAL_FILE_FORMAT.md]: There's a "he" in line 200.
Maybe use "it" and "its" here? It's about a program after all, not a human.

**Review by keszybz — CHANGES_REQUESTED** [review]

File: docs/JOURNAL_FILE_FORMAT.md (hunk 3 above)
Comment (ghost) [reply docs/JOURNAL_FILE_FORMAT.md]: Done.

**Comment by PhoenixDiscord** [comment]
The CI checks have passed successfully.

**Pull request merged**